src/test/java/org/opensearch/ad/task/ADTaskCacheManagerTests.java

```diff
@@ -29,16 +64,17 @@ public void setUp() throws Exception {
             .put(AnomalyDetectorSettings.MAX_CACHED_DELETED_TASKS.getKey(), 100)
             .build();
 
-        clusterService = mock(ClusterService.class);
-        ClusterSettings clusterSettings = new ClusterSettings(
-            settings,
-            Collections
-                .unmodifiableSet(
-                    new HashSet<>(
-                        Arrays.asList(AnomalyDetectorSettings.MAX_BATCH_TASK_PER_NODE, AnomalyDetectorSettings.MAX_CACHED_DELETED_TASKS)
-                    )
-                )
-        );
+        clusterService = mock(SDKClusterService.class);
+        List<Setting<?>> settingsList = List
+            .of(AnomalyDetectorSettings.MAX_BATCH_TASK_PER_NODE, AnomalyDetectorSettings.MAX_CACHED_DELETED_TASKS);
+
+        ExtensionsRunner mockRunner = mock(ExtensionsRunner.class);
+        Extension mockExtension = mock(Extension.class);
+        when(mockRunner.getEnvironmentSettings()).thenReturn(settings);
+        when(mockRunner.getExtension()).thenReturn(mockExtension);
+        when(mockExtension.getSettings()).thenReturn(settingsList);
+        SDKClusterService.SDKClusterSettings clusterSettings = new SDKClusterService(mockRunner).getClusterSettings();
```

Comment:
Nit: SDKClusterService import should be at top. Just my personal opinion. Approving it.